From 9448e86cf44103e5c67a177f7c4116d1e9e9292e Mon Sep 17 00:00:00 2001
From: Christian Wieninger <cwieninger@gmx.de>
Date: Sun, 1 Sep 2013 15:17:10 +0200
Subject: [PATCH 4/4] fix for pin patch

---
 menu_myedittimer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/menu_myedittimer.c b/menu_myedittimer.c
index becce85..a6529a4 100644
--- a/menu_myedittimer.c
+++ b/menu_myedittimer.c
@@ -128,7 +128,7 @@ void cMenuMyEditTimer::Set()
     Add(new cMenuEditStrItem( tr("Directory"), directory, MaxFileName, tr(AllowedChars)));
     Add(new cMenuEditBitItem( trVDR("Active"),       &flags, tfActive));
 #ifdef USE_PINPLUGIN
-    if (cOsd::pinValid) Add(new cMenuEditChanItem(tr("Channel"), &channel));
+    if (cOsd::pinValid || !fskProtection) Add(new cMenuEditChanItem(tr("Channel"), &channel));
     else {
       cString buf = cString::sprintf("%s\t%s", tr("Channel"), Channels.GetByNumber(channel)->Name());
       Add(new cOsdItem(buf));
-- 
1.8.1.5

